src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -1,359 +1,243 @@
-    public static final int POSITION_OF_SERVLETNAME = 1;
-    public static final int POSITION_OF_ACTION = 2;
-    public static final int POSITION_OF_TOSHOW = 3;
-    public static final int POSITION_OF_ROLE = 4;
-    public static final int POSITION_OF_NAME = 5;
-    public static final int POSITION_OF_GOOGLEID = 6;
-    public static final int POSITION_OF_EMAIL = 7;
-    public static final int POSITION_OF_MESSAGE = 8;
-    public static final int POSITION_OF_URL = 9;
-    public static final int POSITION_OF_ID = 10;
-    public static final int POSITION_OF_TIMETAKEN = 11;
-    
+    // TEAMMATESLOG|||ACTION_NAME|||ACTION_RESPONSE|||TO_SHOW|||ROLE|||NAME|||GOOGLE_ID|||EMAIL
+    // |||MESSAGE(IN HTML)|||URL|||TIME_TAKEN
+    public static final int POSITION_OF_ACTION_NAME = 1;
+    public static final int POSITION_OF_ACTION_RESPONSE = 2;
+    public static final int POSITION_OF_LOG_TOSHOW = 3;
+    public static final int POSITION_OF_USER_ROLE = 4;
+    public static final int POSITION_OF_USER_NAME = 5;
+    public static final int POSITION_OF_USER_GOOGLEID = 6;
+    public static final int POSITION_OF_USER_EMAIL = 7;
+    public static final int POSITION_OF_LOG_MESSAGE = 8;
+    public static final int POSITION_OF_ACTION_URL = 9;
+    public static final int POSITION_OF_LOG_ID = 10;
+    public static final int POSITION_OF_LOG_TIMETAKEN = 11;
+    
+    // UI part
     private static final int TIME_TAKEN_WARNING_LOWER_RANGE = 10000;
     private static final int TIME_TAKEN_WARNING_UPPER_RANGE = 20000;
     private static final int TIME_TAKEN_DANGER_UPPER_RANGE = 60000;
     
-    private static final Logger log = Logger.getLogger();
-
-    private long time;
-    private String servletName;
-    private String action; //TODO: remove if not needed (and rename servletName to action)
-    private String role;
-    private String name;
-    private String googleId;
-    private String email;
-    private boolean toShow;
-    private String message;
-    private String url;
-    private Long timeTaken;
+    // Required fields
     
     // id can be in the form of <googleId>%<time> e.g. bamboo3250%20151103170618465
     // or <studentemail>%<courseId>%<time> (for unregistered students)
     //     e.g. bamboo@gmail.tmt%instructor.ema-demo%20151103170618465
-    private String id;
-    private boolean isFirstRow;
+    private String logId;
+    private long logTime;
+    private String actionUrl;
+    private String actionName;
+    
+    // Optional fields
+    
+    private String actionResponse;
+    private String userRole;
+    private boolean isMasqueradeUserRole;
+    
+    private String userName;
+    private String userEmail;
+    private String userGoogleId;
+    
+    private String logMessage;
     
-    @SuppressWarnings("unused") // used by js
-    private String logInfoAsHtml;
+    private long actionTimeTaken;
+    
+    // legacy of messing up UI and logic
```

Comment:
You can see there is a magic field in the log message.
`TEAMMATESLOG|||...|||true|||...`. This field is actually true for all logs.
Previously they use this field to control the displaying of activitlyLog in `AdminActivitLogPageAction`. It is just UI thing, not relevant to the log itself.

The thing is that we cannot remove it for compatibility reason.